File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int maxReach = 0;
        int n = nums.size();

        for (int i = 0; i < n; ++i) {
            if (i > maxReach) {
                return false; // Cannot reach this index
            }
            maxReach = max(maxReach, i + nums[i]);
            if (maxReach >= n - 1) {
                return true; // Can reach or go beyond last index
            }
        }
        return true;
    }
};

File: 0055-jump-game/0055-jump-game.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        mr = 0
        for i in range(len(nums)):
            if(i>mr):
                return False

            mr = max(mr, i+nums[i])

            if(mr>len(nums)-1):
                return True

        return True